v2/pkg/engine/plan/visitor.go

```diff
@@ -1392,3 +1397,103 @@ func (v *Visitor) resolveFetchDependencies(fetchID int) []resolve.FetchDependenc
 	}
 	return dependencies
 }
+
+func (v *Visitor) buildFetchReasons(fetchID int) []resolve.FetchReason {
+	fields, ok := v.plannerFields[fetchID]
+	if !ok {
+		return nil
+	}
+	dsConfig := v.planners[fetchID].DataSourceConfiguration()
+
+	type typedField struct {
+		typeName string
+		field    string
+	}
+	reasons := make([]resolve.FetchReason, 0, len(fields))
+	index := make(map[typedField]int, len(fields))
+
+	for _, fieldRef := range fields {
+		fieldName := v.Operation.FieldNameString(fieldRef)
+		if fieldName == "__typename" {
+			continue
+		}
+		typeName := v.fieldEnclosingTypeNames[fieldRef]
+		if !dsConfig.RequiresFetchReason(typeName, fieldName) {
+			continue
+		}
+
+		byUser := !v.skipField(fieldRef)
+		dependants, ok := v.fieldRefDependants[fieldRef]
+
+		var subgraphs []string
+		var isKey, isRequires bool
+
+		if ok {
```

Comment:
should ok be `hasDependants` or `dependants, ok := v.fieldRefDependants[fieldRef]` be moved right before this line?

Reply:
Yep, left over from old revisions. I have moved this assigment right into the if condition.